src/onnx/onnx.cpp

```diff
@@ -1499,12 +1499,37 @@ struct onnx_parser
             }
             else
             {
-                assert(node.output().size() >= result.size());
-                std::transform(result.begin(),
-                               result.end(),
-                               node.output().begin(),
-                               std::inserter(instructions, instructions.end()),
-                               [](auto&& x, auto&& y) { return std::make_pair(y, x); });
+                std::vector<instruction_ref> vec_ins_refs;
+                auto size = std::min<std::size_t>(node.output().size(), result.size());
+                if(size < result.size())
+                {
+                    for(std::size_t i = size; i < result.size(); ++i)
+                    {
+                        vec_ins_refs.push_back(result.at(i));
+                    }
+                }
+
+                for(auto i = 0; i < size; ++i)
+                {
+                    auto&& x = result.at(i);
+                    auto&& y = node.output()[i];
+                    if(!y.empty())
+                    {
+                        instructions[y] = x;
+                    }
+                    else
+                    {
+                        vec_ins_refs.push_back(x);
+                    }
+                }
+
+                for(auto ins : vec_ins_refs)
+                {
+                    if(ins->outputs().empty())
+                    {
+                        prog.remove_instruction(ins);
```

Comment:
We should rely on DCE to remove the unused outputs.

Reply:
If the ins to be removed is the last ins of the program, DCE will not work.

Reply:
Ok we should probably add `identity` at the end of the program to return the final output(at least until we support multiple returns).